src/overlays/actors/ovl_En_Bigpo/z_en_bigpo.c

```diff
@@ -94,161 +151,1302 @@ static DamageTable D_80B65044 = {
+        }
+        magnitude = magnitude / this->idleTimer;
+        this->actor.world.pos.x -= magnitude * posDiff.x;
+        this->actor.world.pos.y -= magnitude * posDiff.y;
+        this->actor.world.pos.z -= magnitude * posDiff.z;
+    }
+}
+
+void EnBigpo_UpdateColor(EnBigpo* this) {
+    s32 bplus5;
+    s32 bminus5;
+
+    if (this->actionFunc == EnBigpo_CheckHealth) {
+        if (this->actor.colorFilterTimer & 2) {
+            this->mainColor.r = 0;
+            this->mainColor.g = 0;
+            this->mainColor.b = 0;
+        } else {
+            this->mainColor.r = 0x50; // teal? what about the red health?
+            this->mainColor.g = 0xFF;
+            this->mainColor.b = 0xE1;
+        }
+    } else {
+        this->mainColor.r = CLAMP_MAX(this->mainColor.r + 5, 0xFF);
+        this->mainColor.g = CLAMP_MAX(this->mainColor.g + 5, 0xFF);
+
+        // this might be a triple ternary but it matches and is easier to read spread out
+        bplus5 = this->mainColor.b + 5;
+        if (this->mainColor.b >= 0xD3) {
+            bminus5 = this->mainColor.b - 5;
+            if (bminus5 < 0xD2) {
+                this->mainColor.b = 0xD2;
+            } else {
+                this->mainColor.b = bminus5;
+            }
+        } else {
+            if (bplus5 >= 0xD3) {
+                this->mainColor.b = 0xD2;
+            } else {
+                this->mainColor.b = bplus5;
+            }
+        }
+    }
+}
+
+void EnBigpo_FlickerLanternLight(EnBigpo* this) {
+    f32 rand = Rand_ZeroOne();
+    this->lanternColor.r = ((s32)(rand * 30.0f)) + 0xE1;
+    this->lanternColor.g = ((s32)(rand * 100.0f)) + 0x9B;
+    this->lanternColor.b = ((s32)(rand * 160.0f)) + 0x5F;
+    this->lanternColor.a = ((s32)(rand * 30.0f)) + 0xDC;
+}
+
+s32 EnBigpo_ApplyDamage(EnBigpo* this, GlobalContext* globalCtx) {
+    if ((this->collider.base.acFlags & AC_HIT) && !(this->collider.base.acFlags & AC_HARD)) {
+        this->collider.base.acFlags &= ~AC_HIT;
+
+        if (this->actor.colChkInfo.damageEffect == 0xF) {
+            return true;
+        }
+
+        if (Actor_ApplyDamage(&this->actor) == 0) {
+            this->actor.flags &= ~0x1; // targetable OFF
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_DEAD);
+            Enemy_StartFinishingBlow(globalCtx, &this->actor);
+            if (this->actor.params == ENBIGPO_SUMMONED) { // dampe type
+                func_801A2ED8();
+            }
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_PO_DAMAGE);
+        }
+
+        // light arrows
+        if (this->actor.colChkInfo.damageEffect == 4) {
+            this->unk21C = 4.0f;
+            this->unk220 = 1.0f;
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->collider.info.bumper.hitPos.x,
+                        this->collider.info.bumper.hitPos.y, this->collider.info.bumper.hitPos.z, 0, 0, 0, 
+                        CLEAR_TAG_EFFECT_FLASH);
```

Comment:
```suggestion
            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->collider.info.bumper.hitPos.x,
                        this->collider.info.bumper.hitPos.y, this->collider.info.bumper.hitPos.z, 0, 0, 0, 
                        CLEAR_TAG_LARGE_LIGHT_RAYS);
```
Errr, this is the enums you need:
https://github.com/zeldaret/mm/blob/1c544d235172053de78e568798df1cd40ba06688/include/z64actor.h#L1007

Reply:
oh, oops